R/RunQuery.R

```diff
@@ -25,14 +28,33 @@
 #' result <- RunQuery(query, df)
 #'
 #' @export
-RunQuery <- function(strQuery, df) {
+RunQuery <- function(strQuery, df, bUseSchema = FALSE, lColumnMapping = NULL) {
   stopifnot(is.character(strQuery))
 
   # Check that strQuery contains "FROM df"
   if (!stringr::str_detect(strQuery, "FROM df")) {
     stop("strQuery must contain 'FROM df'")
   }
 
+  # Check that columnMapping exists if use_schema == TRUE
+
+  if (bUseSchema && is.null(lColumnMapping)) {
+    stop("if use_schema = TRUE, you must provide lColumnMapping spec")
+  }
+
+  #use `source_col` for `source` if using mapping and it hasn't gone through ApplySpec()
+  if (bUseSchema && any(is.null(map(lColumnMapping, \(x) x$source)))) {
```

Comment:
```suggestion
  if (bUseSchema && any(is.null(map_lgl(lColumnMapping, \(x) is.null(x$source)))) {
```

Reply:
@samussiah your adjustment also needs to remove then the global is.null - else is.null will nullify any of the logical checks so it will always return false since everything will be true/false

Reply:
ah i think @lauramaxwell already picked up on this

Reply:
Yeah, I accidentally left in the outer `is.null`.